cupy/_creation/matrix.py

```diff
@@ -141,7 +141,35 @@ def triu(m, k=0):
     return cupy.where(mask, m.dtype.type(0), m)
 
 
-# TODO(okuta): Implement vander
+def vander(x, N=None, increasing=False):
+    """Returns a Vandermonde matrix.
+
+    Args:
+        x (array-like): 1-D array or array-like object.
+        N (int, optional): Number of columns in the output.
+            ``N = len(x)`` by default.
+        increasing (bool, optional): Order of the powers of the columns.
+            If True, the powers increase from right to left,
+            if False (the default) they are reversed.
+
+    Returns:
+        cupy.ndarray: A Vandermonde matrix.
+
+    .. seealso:: :func:`numpy.vander`
+
+    """
+    x = cupy.asarray(x)
```

Comment:
Hi, @toslunar, @asi1024, @tom24d,
I was very interested in looking into the PR and took a glance at it. I wonder if we can modify this statement to
```
if not isinstance(x, cupy.ndarray):
    raise TypeError('`x` should be of type cupy.ndarray')
```
To avoid host-to-device sync as mentioned in https://github.com/cupy/cupy/pull/6280#discussion_r779972513. Benefits of it:
- Performance of the model will be affected if we apply internal synchronization without the user's knowledge.

Credits: @takagi! Thanks to you (:

Reply:
@khushi-411 In our policy, the features classified as [Array creation routines](https://numpy.org/doc/stable/reference/routines.array-creation.html) (e.g., `vander`, `asarray`, `diag`, ...) should allow all ArrayLike inputs.  The details of the policy are still under discussion, but we are planning to update the [Development policies](https://github.com/cupy/cupy/wiki/Development-policies) after they are finalized!

Reply:
Oo, got it! Thanks for it (: @asi1024!
(I would love to read the updated policies :))